endpoint/endpoint.c

```diff
@@ -64,7 +70,8 @@ status_t send_transaction_information(int value, const char* message, const char
     fprintf(stderr, "%s\n", "encrypt msg error");
     return ret;
   }
-  serialize_msg(iv, encrypt_ctx.ciphertext_len, (char*)encrypt_ctx.ciphertext, msg, &msg_len);
+  serialize_msg(iv, encrypt_ctx.ciphertext_len, (char*)encrypt_ctx.ciphertext, encrypt_ctx.timestamp, encrypt_ctx.hmac,
```

Comment:
Would you consider to pass `encrypt_ctx` instead of its member data?
Cc. @HowJMay

Reply:
I think it is good idea to do so

Reply:
Let's create a new issue to refactor. I defer to @HowJMay to determinate the approval of this pull request though.

Reply:
Create issue at #627